pkg/workload/workload.go

```diff
@@ -205,17 +204,10 @@ func UpdateStatus(ctx context.Context,
 		Reason:             reason,
 		Message:            api.TruncateConditionMessage(message),
 	}
-	// Avoid modifying the object in the cache.
-	newWl := *wl
-	newWl.Status = *newWl.Status.DeepCopy()
-
-	if conditionIndex == -1 {
-		newWl.Status.Conditions = append(newWl.Status.Conditions, condition)
-	} else {
-		newWl.Status.Conditions[conditionIndex] = condition
-	}
 
-	return c.Status().Update(ctx, &newWl)
+	newWl := ClearAdmissionPatch(wl)
```

Comment:
This function needs a name change now. It's no longer about admission.

Maybe `BaseSSAWorkload`?